app/controllers/vendor_api/references_controller.rb

```diff
@@ -0,0 +1,44 @@
+module VendorAPI
+  class ReferencesController < VendorAPIController
+    before_action :sandbox_only
+
+    def success
+      with_requested_reference do |reference|
+        reference.update!(
+          feedback_status: 'feedback_provided',
+          feedback_provided_at: Time.zone.now,
+          feedback: Faker::Lorem.paragraph(sentence_count: 10),
+          safeguarding_concerns: '',
+          relationship_correction: '',
+        )
+      end
+    end
+
+    def failure
+      with_requested_reference do |reference|
+        reference.update!(
+          feedback_status: 'feedback_refused',
+          feedback_refused_at: Time.zone.now,
+        )
+      end
+    end
+
+  private
+
+    def sandbox_only
```

Comment:
do we know if `HostingEnvironment.sandbox_mode?` returns true for QA and review apps? i think we want this functionality to be available on QA. perhaps the correct test here is `HostingEnvironment.sandbox_mode? || HostingEnvironment.test_environment?`, or simply `!HostingEnvironment.production?`.

Reply:
@carlosmartinez Updated.